debezium-embedded/src/test/java/io/debezium/embedded/async/AbstractAsyncEngineConnectorTest.java

```diff
@@ -17,18 +16,17 @@
  */
 public class AbstractAsyncEngineConnectorTest extends AbstractConnectorTest {
 
-    protected DebeziumEngine.Signaler<SignalRecord> signaler;
-
     @Override
     protected DebeziumEngine.Builder createEngineBuilder() {
-        this.signaler = new AsyncEngineSignaler();
-
-        return new AsyncEmbeddedEngine.AsyncEngineBuilder()
-                .using(signaler);
+        return new AsyncEmbeddedEngine.AsyncEngineBuilder();
     }
 
     @Override
     protected TestingDebeziumEngine createEngine(DebeziumEngine.Builder builder) {
         return new TestingAsyncEmbeddedEngine((AsyncEmbeddedEngine) builder.build());
     }
+
+    protected DebeziumEngine.Signaler getSignaler() {
+        return (DebeziumEngine.Signaler) engine.getSignaler().get();
```

Comment:
This needs to be here as the entirety of Test infra uses generics poorly (the whole things shines with yellow underlines like a Christmas tree)  

I will try to fix it tomorrow but it might be a long way down the rabbit hole. It alls stems from the fact that the following 


```java
static class Foo<T> {
    Optional<String> getName() {
        return Optional.of("John Doe");
    }
}

public static void main(String[] args) {
    Foo foo = new Foo();
    String name = foo.getName().get();
    System.out.println(name);
}
```

will not compile. As ALL generic information is erased on `foo` thus `name` is of type `Object`. 

Our entire suite uses raw types instead of `<?>` where it doesn't care about the generic type.

Reply:
Filed https://issues.redhat.com/browse/DBZ-8166